### PR TITLE
ci(github-actions): support continuous deployment to multiple environments

### PR DESCRIPTION
**Pull request by Paz-X**
Merging the CD portion of the Github Actions.
We are able build and push to the appropriate S3 bucket
This resolves ghi #3

### CODE REVIEW

File: .github/workflows/BuildNDeployDev.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# This workflow will checkout the contents of the `S3Dev` branch of this repo, remove all files in the current S3 bucket, and push the contents
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml:1]: `S3Dev` -> `development` branch

File: .github/workflows/BuildNDeployDev.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+# to the bucket. Once the contents of the S3 bucket has been updated, a call will be made to invalidate the cache in the CloudFront distribution.
+# The CloudFront distribution is configured to pull the contents from the S3 bucket once an `invalidation` request is sent.
+
+name: Build and Deploy Static Site
+
+on: # run this workflow when a push has been made to `Jekyll-Alf` branch
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: update comment to reflect `development` branch

File: .github/workflows/BuildNDeployDev.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  push:
+    branches:
+      - Jekyll-Alf
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: branch will be `development`.

File: .github/workflows/BuildNDeployDev.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: hard fix this to `ubuntu-20.04`

File: .github/workflows/BuildNDeployDev.yml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+          run: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle
+          
+        - name: Install Jekyll-Alf
+          run: bundle exec jekyll build --config _config.yml,_config_prod.yml
+          
+          
+          #Mmm, trying to clean dir as in CircleCI Config file but not the same
+        #- name: Clean temp directory
+         # run: rm -rf *
+          #working-directory: _site/
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: lets remove this.

File: .github/workflows/BuildNDeployDev.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          with:
+            ruby-version: '2.6' # Version range or exact version of a Ruby version to use, using semvers version range syntax.
+       
+
+        - uses: actions/cache@v2
+          with:
+            path: vendor/bundle
+            key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+            restore-keys: |
+              ${{ runner.os }}-gems-
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml:33]: Add comment section to explain a bit more detail about this.

File: .github/workflows/BuildNDeployDev.yml [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# This workflow will checkout the contents of the `S3Dev` branch of this repo, remove all files in the current S3 bucket, and push the contents
+# to the bucket. Once the contents of the S3 bucket has been updated, a call will be made to invalidate the cache in the CloudFront distribution.
+# The CloudFront distribution is configured to pull the contents from the S3 bucket once an `invalidation` request is sent.
+
+name: Build and Deploy Static Site
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: `Build and Deploy Static site to Dev Environment`

File: .github/workflows/BuildNDeployDev.yml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+        - uses: actions/cache@v2
+          with:
+            path: vendor/bundle
+            key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+            restore-keys: |
+              ${{ runner.os }}-gems-
+    
+        #- name: Running basic Ruby script
+         # run: ruby hello.rb
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: let's remove these comments as well.

**Review by jaylenw — COMMENTED** [review]
The comments I mentioned for the dev yml workflow file, let's address the same for the prod.

File: _config.yml [hunk 9]
```diff
@@ -1,124 +1,127 @@
...
+# -- https://github.com/my-awesome-github-org/my-awesome-data-repo
+# Then you would want to make these replacements throughout this file:
+# -- replace "MY-GITHUB-ORG" with "my-awesome-github-org"
+# -- replace "OPEN-SDG-SITE-STARTER" with "my-awesome-site-repo"
+# -- replace "OPEN-SDG-DATA-STARTER" with "my-awesome-data-repo"
+
+# Replace this title as needed.
+title: City of Los Angeles Indicators For The Sustainable Development Goals
+# Replace the baseurl with the name of this repository.
+baseurl: "/open-sdg-site-starter"
```
Comment (jaylenw) [inline _config.yml]: baseurl should now be `""`, like the way it is in `_config_prod.yml`

File: _config.yml [hunk 10]
```diff
@@ -1,124 +1,127 @@
...
+url: ""
+data_dir: data
+environment: staging
```
Comment (jaylenw) [inline _config.yml]: `environment: development`

File: .github/workflows/BuildNDeployDev.yml [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        
+        - name: Install Ruby dependencies
+          run: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle
+          
+        - name: Install Jekyll-Alf
+          run: bundle exec jekyll build --config _config.yml,_config_prod.yml
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: Just for dev, remove `_config_prod.yml`

File: .github/workflows/BuildNDeployDev.yml [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+            restore-keys: |
+              ${{ runner.os }}-gems-
+    
+        #- name: Running basic Ruby script
+         # run: ruby hello.rb
+        
+        - name: Install Ruby dependencies
+          run: bundle check --path=vendor/bundle || bundle install --path=vendor/bundle
+          
+        - name: Install Jekyll-Alf
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: Let's provide a more appropriate name.

File: .github/workflows/BuildNDeployDev.yml [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+ ###########################################################################################################
+ #      This is the CI portion
+ ###########################################################################################################
+        - uses: actions/checkout@v2
+        - uses: actions/setup-ruby@v1
+          with:
+            ruby-version: '2.6' # Version range or exact version of a Ruby version to use, using semvers version range syntax.
+       
+
+        - uses: actions/cache@v2
```
Comment (jaylenw) [inline .github/workflows/BuildNDeployDev.yml]: Provide names for these so when we look at our progress in the Actions dashboard, we know what is exactly running for a given step.

File: .github/workflows/BuildNDeployDev.yml (hunk 6 above)
Comment (jaylenw) [reply .github/workflows/BuildNDeployDev.yml:33]: @Paz-X  create an issue to come back and add appropriate comments as we discussed.

**Review by jaylenw — APPROVED** [review]
Approved with the latest changes.

**Review by RV-LACity — APPROVED** [review]
All looks good.

**Pull request merged**